polkadb/database_test.go

```diff
@@ -0,0 +1,117 @@
+package polkadb
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func newTestBadgerDB() (*BadgerDB, func()) {
+	dir, err := ioutil.TempDir(os.TempDir(), "badger-test")
+	fmt.Println(dir)
+	if err != nil {
+		panic("failed to create test file: " + err.Error())
+	}
+	db, err := NewBadgerDB(dir)
+	if err != nil {
+		panic("failed to create test database: " + err.Error())
+	}
+	return db, func() {
+		db.Close()
+		os.RemoveAll(dir)
+	}
+}
+
+func TestBadgerDB_PutGetDel(t *testing.T) {
+	db, remove := newTestBadgerDB()
+	defer remove()
+	testPutGetter(db, t)
+	testDelGet(db, t)
+}
+
+func testPutGetter(db Database, t *testing.T) {
+	t.Parallel()
```

Comment:
Any advice when/when not to use?

Reply:
From my experience, if you're using a number of ranges or loops it's certainly useful. However, they're some common gotcha's...... the primary being detailed in this gist https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721